hermes/src/lib.rs

```diff
@@ -62,27 +62,32 @@ pub trait ComponentFacade: Send + Sync {
     fn publish_version_request(&self) -> Fallible<()>;
     fn subscribe_version(&self, handler: Callback<VersionMessage>) -> Fallible<()>;
     fn subscribe_error(&self, handler: Callback<ErrorMessage>) -> Fallible<()>;
+    fn subscribe_loaded(&self, handler: Callback<LoadMessage>) -> Fallible<()>;
 }
 
 /// A generic facade used to interact with a component
 pub trait IdentifiableComponentFacade: Send + Sync {
     fn publish_version_request(&self, id: String) -> Fallible<()>;
     fn subscribe_version(&self, id: String, handler: Callback<VersionMessage>) -> Fallible<()>;
     fn subscribe_error(&self, id: String, handler: Callback<ErrorMessage>) -> Fallible<()>;
+    fn subscribe_loaded(&self, id: String, handler: Callback<SiteLoadMessage>) -> Fallible<()>;
+    fn subscribe_all_loaded(&self, handler: Callback<SiteLoadMessage>) -> Fallible<()>;
 }
 
 /// A generic facade all components must use to publish their errors and versions (when requested)
 pub trait ComponentBackendFacade: Send + Sync {
     fn subscribe_version_request(&self, handler: Callback0) -> Fallible<()>;
     fn publish_version(&self, version: VersionMessage) -> Fallible<()>;
     fn publish_error(&self, error: ErrorMessage) -> Fallible<()>;
+    fn publish_loaded(&self, loaded: LoadMessage) -> Fallible<()>;
 }
 
 /// A generic facade all components must use to publish their errors and versions (when requested)
 pub trait IdentifiableComponentBackendFacade: Send + Sync {
     fn subscribe_version_request(&self, id: String, handler: Callback0) -> Fallible<()>;
     fn publish_version(&self, id: String, version: VersionMessage) -> Fallible<()>;
     fn publish_error(&self, id: String, error: ErrorMessage) -> Fallible<()>;
+    fn publish_loaded(&self, id: String, loaded: SiteLoadMessage) -> Fallible<()>;
```

Comment:
as the `site_id` is present in the `SiteLoadMessage` lets not require to put it here (we should fix that for the other methods here, but certainly in another PR)